Spec/RealtimeClient.swift

```diff
@@ -352,6 +352,478 @@ class RealtimeClient: QuickSpec {
+                            done()
+                        }
+                        client.connect()
+                    }
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.auth.authorize(nil, options: nil) { tokenDetails, error in
+                            expect(error).to(beNil())
+                            expect(tokenDetails).toNot(beNil())
+
+                            guard let transport = client.transport as? TestProxyTransport else {
+                                fail("TestProxyTransport is not set"); done(); return
+                            }
+
+                            expect(transport.protocolMessagesSent.filter({ $0.action == .Auth })).to(haveCount(1))
+                            expect(transport.protocolMessagesReceived.filter({ $0.action == .Connected })).to(haveCount(2))
+                            expect(transport.protocolMessagesReceived.filter({ $0.action == .Error })).to(haveCount(0))
+                            done()
+                        }
+                    }
+                }
+
+                // RTC8b
+                it("when connection is CONNECTING, all current connection attempts should be halted, and after obtaining a new token the library should immediately initiate a connection attempt using the new token") {
+                    let options = AblyTests.commonAppSetup()
+                    options.autoConnect = false
+                    options.useTokenAuth = true
+                    let client = ARTRealtime(options: options)
+                    defer { client.dispose(); client.close() }
+                    client.setTransportClass(TestProxyTransport.self)
+
+                    waitUntil(timeout: testTimeout) { done in
+                        client.connection.once(.Connecting) { stateChange in
+                            expect(stateChange?.reason).to(beNil())
+
+                            let authOptions = ARTAuthOptions()
+                            authOptions.key = AblyTests.commonAppSetup().key
+
+                            client.auth.authorize(nil, options: authOptions) { tokenDetails, error in
+                                expect(error).to(beNil())
+                                guard let tokenDetails = tokenDetails else {
+                                    fail("TokenDetails is nil"); done(); return
+                                }
+                                expect(tokenDetails.token).toNot(beNil())
+                                expect(client.connection.state).to(equal(ARTRealtimeConnectionState.Connected))
+
+                                guard let transport = client.transport as? TestProxyTransport else {
+                                    fail("TestProxyTransport is not set"); done(); return
+                                }
+                                expect(transport.protocolMessagesReceived.filter({ $0.action == .Connected })).to(haveCount(1))
```

Comment:
This seems fine, but I'm not sure how you are confirming that the original connection attempt is cancelled.  I wouldn't worry too much about it, but I am not sure I see how this tests confirms that?